File: Signiture-Backend/kotlin/delivery/db/src/main/java/pcy/study/db/user/UserRepository.java
package pcy.study.db.user;

import org.springframework.data.jpa.repository.JpaRepository;
import pcy.study.db.user.enums.UserStatus;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {

    Optional<User> findFirstByIdAndStatusOrderByIdDesc(Long id, UserStatus userStatus);

    Optional<User> findFirstByEmailAndPasswordAndStatusOrderByIdDesc(String email, String password, UserStatus status);
}

File: Signiture-Backend/kotlin/delivery/db/src/main/kotlin/pcy/study/db/user/UserRepository.kt
package pcy.study.db.user

import org.springframework.data.jpa.repository.JpaRepository
import pcy.study.db.user.enums.UserStatus

interface UserRepository: JpaRepository<User, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(
        id: Long?,
        userStatus: UserStatus?
    ): User?

    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): User?
}